package/MDAnalysis/core/topologyattrs.py

```diff
@@ -651,6 +651,20 @@ def _gen_initial_values(na, nr, ns):
         return np.zeros(na)
 
 
+class RecordTypes(AtomAttr):
+    """For PDB-like formats, indicates if ATOM or HETATM
+
+    Defaults to 'ATOM'
+    """
+    attrname = 'record_types'
+    singular = 'record_type'
+    per_object = 'atom'
+
+    @staticmethod
+    def _gen_initial_values(na, nr, ns):
+        return np.array(['ATOM'] * na, dtype=object)
```

Comment:
It seems (memory)-wasteful to store a string with each atom. For 1M atoms, that can be an additional `len("HETATM") * 1e6 / (1024*1024)` = 5.7 MiB. Could we instead use a small integer, such as 1 = ATOM, 2 = HETATM, 0 = no idea, and then get the string back on the fly (e.g., as a dict look-up)?

Or is this more complicated than the memory consumption? 

I'm just conscientious of the fact that we will be using the additional memory for *any* PDB file (especially simulation systems), where no-one cares about the atom records per se.